lib/polo/sql_translator.rb

```diff
@@ -7,47 +9,35 @@ class SqlTranslator
     def initialize(object, configuration=Configuration.new)
       @record = object
       @configuration = configuration
-    end
-
-    def to_sql
-      records = Array.wrap(@record)
 
-      sqls = records.map do |record|
-        raw_sql(record)
-      end
-
-      if @configuration.on_duplicate_strategy == :ignore
-        sqls = ignore_transform(sqls)
+      case @configuration.adapter
+      when :mysql then self.extend(Polo::Adapters::MySQL)
+      when :postgres then self.extend(Polo::Adapters::Postgres)
+      else
+        raise "Unknown SQL adapter: #{@configuration.adapter}"
       end
+    end
 
-      if @configuration.on_duplicate_strategy == :override
-        sqls = on_duplicate_key_update(sqls, records)
+    def to_sql
+      case @configuration.on_duplicate_strategy
+      when :ignore then ignore_transform(inserts, records)
```

Comment:
as mentioned above.. this would become `@adapter.ignore_transform(inserts, records)`